src/Oscoin/Storage/Block/SQLite/Tx.hs

```diff
@@ -0,0 +1,162 @@
+    , getTx
+    , getBlockTxs
+    , storeTxs
+    ) where
+
+import           Oscoin.Prelude
+
+import           Oscoin.Crypto.Blockchain.Block (BlockHash)
+import           Oscoin.Crypto.Hash (Hash, Hashable, fromHashed, hash)
+import           Oscoin.Crypto.Hash.RealWorld ()
+import           Oscoin.Crypto.PubKey
+import           Oscoin.Crypto.PubKey.RealWorld ()
+import           Oscoin.Data.Ledger (AccountId, Balance, Nonce)
+import           Oscoin.Data.OscoinTx
+
+import           Codec.Serialise
+import qualified Data.ByteString.Lazy as LBS
+
+import           Database.SQLite.Simple as Sql (Connection, Only(..))
+import qualified Database.SQLite.Simple as Sql
+import           Database.SQLite.Simple.FromField (FromField)
+import           Database.SQLite.Simple.FromRow (FromRow, field)
+import           Database.SQLite.Simple.Orphans ()
+import           Database.SQLite.Simple.QQ
+import           Database.SQLite.Simple.ToField (ToField, toField)
+
+-- | Constraints required for a @Tx c@ to be storable in SQLite.
+type StorableTx c =
+    ( Serialise (Signature c)
+    , Serialise (AccountId c)
+    , Serialise (PublicKey c)
+    , Serialise (Hash c)
+    , FromField (Signature c)
+    , FromField (Hash c)
+    , ToField (Signature c)
+    , ToField (Hash c)
+    , Hashable c (Tx c)
+    )
+
+data TxRow c = TxRow
+    { txRowHash      :: Hash c
+    , txRowNetwork   :: Word8
+    , txRowSignature :: Signature c
+    , txRowMessages  :: ByteString
+    , txRowNonce     :: Nonce
+    , txRowFee       :: Balance
+    , txRowBurn      :: Balance
+    , txRowAuthor    :: PublicKey c
+    }
+
+instance (StorableTx c, Typeable c) => FromRow (TxRow c) where
+    fromRow = do
+        txRowHash      <- field
+        txRowNetwork   <- field
+        txRowSignature <- field
+        txRowMessages  <- field
+        txRowNonce     <- field
+        txRowFee       <- field
+        txRowBurn      <- field
+        txRowAuthor    <- field
+
+        pure TxRow{..}
+
+
+newtype TxRowDecodeError = TxRowDecodeError String
+    deriving (Show, Eq)
+
+instance Exception TxRowDecodeError where
+    displayException (TxRowDecodeError msg) =
+        "Failed to decode transaction row: " <> msg
+
+toTxRow ::
+    ( StorableTx c
+    ) => Tx c -> TxRow c
+toTxRow tx@Tx'{..} = TxRow
+    { txRowHash      = fromHashed $ hash tx
+    , txRowNetwork   = fromIntegral $ fromEnum txNetwork
+    , txRowSignature = txSignature
+    , txRowMessages  = LBS.toStrict $ serialise (txMessages txPayload)
+    , txRowNonce     = txNonce txPayload
+    , txRowFee       = txFee txPayload
+    , txRowBurn      = txBurn txPayload
+    , txRowAuthor    = txAuthor txPayload
+    }
+
+fromTxRow :: StorableTx c => TxRow c -> Either TxRowDecodeError (Tx c)
+fromTxRow TxRow{..} = first toDecodeError $ do
+    txNetwork   <- pure $ toEnum (fromIntegral txRowNetwork)
+    txSignature <- pure txRowSignature
+    txMessages  <- deserialiseOrFail $ LBS.fromStrict txRowMessages
```

Comment:
Wish list: we should create a `Codec.Serialise.Extended` where we export `deserialise'` and `deserialiseOrFail'` which both takes strict `ByteString`s. We have been doing this extra conversion enough in the codebase to be annoying (and repetitive) 😁 Mind opening a ticket? (this is orthogonal to this code review, obviously!)

Reply:
#638